helper/acctest/skip_on_error.go

```diff
@@ -0,0 +1,20 @@
+package acctest
+
+import (
+	"regexp"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+// SkipOnErrorContains helps skip tests based on a portion of an error message
+// which is matched to an error.
+func SkipOnErrorContains(skipError string) resource.SkipOnErrorFunc {
+	return func(err error) bool {
+		if err == nil {
+			return false
+		}
+
+		re := regexp.MustCompile(skipError)
+		return re.MatchString(err.Error())
```

Comment:
Since this typically will be ran after testing infrastructure is provisioned, a regular expression that does not compile will cause a panic and leave all the dangling infrastructure. This is typically why a lot of the other testing code has uses `regexp.MustCompile()` in the test functions themselves, since it forces a Go build error.

This should either use `strings.Contains()` for safety or move the regexp compilation back into the providers.

Reply:
I added helpers and accompanying tests for both `SkipOnErrorContains()` and `SkipOnErrorRegexp()` which are safe.